downstairs/src/region.rs

```diff
@@ -709,67 +717,112 @@ impl Extent {
+        } else {
+            /*
+             * Create the metadata db
+             */
+            path.set_extension("db");
+            let metadb = open_sqlite_connection(&path)?;
 
-        metadb.execute(
-            "CREATE TABLE integrity_hashes (
-                counter INTEGER,
-                block INTEGER,
-                hash BLOB NOT NULL,
-                PRIMARY KEY (block, counter)
-            )",
-            [],
-        )?;
+            /*
+             * Create tables and insert base data
+             */
+            metadb.execute(
+                "CREATE TABLE metadata (
+                    name TEXT PRIMARY KEY,
+                    value INTEGER NOT NULL
+                )",
+                [],
+            )?;
+
+            let meta = ExtentMeta::default();
+
+            metadb.execute(
+                "INSERT INTO metadata
+                (name, value) VALUES (?1, ?2)",
+                params!["ext_version", meta.ext_version],
+            )?;
+            metadb.execute(
+                "INSERT INTO metadata
+                (name, value) VALUES (?1, ?2)",
+                params!["gen_number", meta.gen_number],
+            )?;
+            metadb.execute(
+                "INSERT INTO metadata (name, value) VALUES (?1, ?2)",
+                params!["flush_number", meta.flush_number],
+            )?;
+            metadb.execute(
+                "INSERT INTO metadata (name, value) VALUES (?1, ?2)",
+                params!["dirty", meta.dirty],
+            )?;
+
+            metadb.execute(
+                "CREATE TABLE encryption_context (
+                    counter INTEGER,
+                    block INTEGER,
+                    nonce BLOB NOT NULL,
+                    tag BLOB NOT NULL,
+                    PRIMARY KEY (block, counter)
+                )",
+                [],
+            )?;
+
+            metadb.execute(
+                "CREATE TABLE integrity_hashes (
+                    counter INTEGER,
+                    block INTEGER,
+                    hash BLOB NOT NULL,
+                    PRIMARY KEY (block, counter)
+                )",
+                [],
+            )?;
+
+            // write out
+            metadb.close().map_err(|e| std::io::Error::new(
+                std::io::ErrorKind::Other,
+                format!("metadb.close() failed! {}", e.1.to_string()),
+            ))?;
+
+            // Save it as DB seed
+            std::fs::copy(&path, &seed)?;
+
+            path.set_extension("db-shm");
+            seed.set_extension("db-shm");
+            if Path::new(&path).exists() {
+                std::fs::copy(&path, &seed)?;
+            }
+
+            path.set_extension("db-wal");
+            seed.set_extension("db-wal");
+            if Path::new(&path).exists() {
+                std::fs::copy(&path, &seed)?;
+            }
+
+            path.set_extension("db");
+            open_sqlite_connection(&path)?
```

Comment:
Is the intent to copy all files? If so, should we be doing a readdir + copy?

Reply:
it's not necessary to copy db-shm and db-wal, they won't exist yet - removed in 9542d11